ouroboros-network-framework/sim-tests/Test/Ouroboros/Network/Server2/Sim.hs

```diff
@@ -346,20 +376,92 @@ instance (Arbitrary peerAddr, Arbitrary req, Ord peerAddr) =>
         event <- frequency $
                     [ (6, StartClient             <$> delay <*> newClient)
                     , (6, StartServer             <$> delay <*> newServer <*> arbitrary) ] ++
-                    [ (4, InboundConnection       <$> delay <*> elements possibleInboundConnections)        | not $ null possibleInboundConnections] ++
-                    [ (4, OutboundConnection      <$> delay <*> elements possibleOutboundConnections)       | not $ null possibleOutboundConnections] ++
-                    [ (6, CloseInboundConnection  <$> delay <*> elements inboundConnections)                | not $ null inboundConnections ] ++
-                    [ (4, CloseOutboundConnection <$> delay <*> elements outboundConnections)               | not $ null outboundConnections ] ++
-                    [ (10, InboundMiniprotocols   <$> delay <*> elements inboundConnections  <*> genBundle) | not $ null inboundConnections ] ++
-                    [ (8, OutboundMiniprotocols  <$> delay <*> elements outboundConnections <*> genBundle) | not $ null outboundConnections ] ++
-                    [ (4, ShutdownClientServer    <$> delay <*> elements possibleStoppable)                 | not $ null possibleStoppable ]
+                    [ (4, genEventWithDelay   InboundConnectionShape)                 | not $ null possibleInboundConnections] ++
+                    [ (4, genEventWithDelay   OutboundConnectionShape)                | not $ null possibleOutboundConnections] ++
+                    [ (6, genEventWithDelay   CloseInboundConnectionShape)            | not $ null inboundConnections ] ++
+                    [ (4, genEventWithDelay   CloseOutboundConnectionShape)           | not $ null outboundConnections ] ++
+                    [ (10, genEventWithBundle InboundMiniprotocolsShape    genBundle) | not $ null inboundConnections ] ++
+                    [ (8, genEventWithBundle  OutboundMiniprotocolsShape   genBundle) | not $ null outboundConnections ] ++
+                    [ (4, genEventWithDelay   ShutdownClientServerShape)              | not $ null possibleStoppable]
         (event :) <$> go (nextState event s) (n - 1)
         where
+          genEventWithDelay :: ConnectionEventShape
+                            -> Gen (ConnectionEvent req peerAddr)
+          genEventWithDelay es@InboundConnectionShape      = do
+            (x1, _) <- elements (connectionMap es)
+            let prevDelay = maybe delay (return . snd)
+                          $ find ((== x1) . fst)
+                          $ connectionMap es
+            d <- max <$> prevDelay <*> delay
```

Comment:
it's strange that we use `max`, as if the delays used where `Time` rather than relative `DiffTime`.

Reply:
Why is it strange? I want to find the largest delay so as to not generate a command that could happen before the previous one has run